src/comm/TcpAcceptor.cc

```diff
@@ -399,12 +399,20 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     Ip::Address::FreeAddr(gai);
 
     // Perform NAT or TPROXY operations to retrieve the real client/dest IP addresses
-    if (conn->flags&(COMM_TRANSPARENT|COMM_INTERCEPTION) && !Ip::Interceptor.Lookup(details, conn)) {
-        debugs(50, DBG_IMPORTANT, "ERROR: NAT/TPROXY lookup failed to locate original IPs on " << details);
-        // Failed.
-        PROF_stop(comm_accept);
-        return Comm::COMM_ERROR;
-    }
+    /* NP: try TPROXY first, its much quieter than NAT when non-matching */
+    if (listenPort_->flags.tproxyInterceptLocally()) {
+        if (!Ip::Interceptor.LookupTproxy(details)) {
+             debugs(50, DBG_IMPORTANT, "ERROR: TPROXY lookup failed to locate original IPs on " << details);
+             PROF_stop(comm_accept);
+             return Comm::COMM_ERROR;
+         }
+    } else if (listenPort_->flags.natInterceptLocally()) {
+        if (!Ip::Interceptor.LookupNat(details)) {
+            debugs(50, DBG_IMPORTANT, "ERROR: NAT lookup failed to locate original IPs on " << details);
+            PROF_stop(comm_accept);
+            return Comm::COMM_ERROR;
+        }
+    } // else the PROXY protocol will supply the real client/dest IP addresses
```

Comment:
This comment does not match the code AFAICT -- there are lots of other possible outcomes here, including no PROXY protocol at all. This can become a part of the opening comment:

```C++
// Handle locallyIntercepted() traffic here. The PROXY protocol supplies original
// client and destination IP addresses later, after parsing the PROXY protocol prefix.
if (listenPort_->flags.tproxyInterceptLocally()) {
```